crates/api/src/entrypoint.rs

```diff
@@ -1,30 +1,157 @@
+        me.sgx_profile = parse_bool(SGX_VAR)?;
+        Ok(me)
+    }
 
-    Ok(())
+    /** Configures the maximum size, in bytes, where a linear memory is
+     * considered static, above which it'll be considered dynamic.
+     */
+    pub fn with_static_memory(mut self, max_mamory: impl Into<Option<u64>>) -> Self {
+        self.max_static_memory = max_mamory.into();
+        self
+    }
+
+    /**
+     * Changes default optimization level.
+     *
+     *  * `true` - optimization for speed.
+     *  * `false` - no optimization.
+     *
+     */
+    pub fn with_optimize(mut self, optimize: bool) -> Self {
+        self.optimize = Some(optimize);
+        self
+    }
+
+    /** Enables configuration for Graphene-SGX.
+     */
+    pub fn with_sgx_profile(mut self, sgx_profile: bool) -> Self {
+        self.sgx_profile = Some(sgx_profile);
+        self
+    }
+
+    pub(crate) fn is_default(&self) -> bool {
+        self.max_static_memory.is_none() && self.optimize.is_none() && self.sgx_profile.is_none()
+    }
+
+    /// Instantiates and executes the deployed image using Wasmtime runtime.
+    pub fn run(
+        self,
+        workdir: impl AsRef<Path>,
+        entrypoint: impl AsRef<str>,
+        args: impl IntoIterator<Item = String>,
+    ) -> Result<()> {
+        let workdir = workdir.as_ref();
+        let deploy_file = DeployFile::load(workdir)?;
+
+        let mut image = WasmImage::new(&deploy_file.image_path())?;
+        let mut wasmtime = create_wasmtime(workdir, &deploy_file, self)?;
+
+        info!(
+            "Running image: {:?}",
+            get_log_path(workdir, deploy_file.image_path())
+        );
+        info!("Running image: {}", deploy_file.image_path().display());
+
+        // Since wasmtime object doesn't live across binary executions,
+        // we must deploy image for the second time, what will load binary to wasmtime.
+        let entrypoint = image.find_entrypoint(entrypoint.as_ref())?;
+        wasmtime.load_binary(&mut image, &entrypoint)?;
+        wasmtime.run(entrypoint, args.into_iter().collect())?;
+
+        info!("Computations completed.");
+
+        Ok(())
+    }
+
+    /// Validates the deployed image.
+    pub fn start(self, workdir: impl AsRef<Path>) -> Result<()> {
+        let workdir = workdir.as_ref();
+        let deploy_file = DeployFile::load(workdir)?;
+
+        info!(
+            "Validating deployed image {:?}.",
+            get_log_path(workdir, deploy_file.image_path())
+        );
+
+        let mut image = WasmImage::new(&deploy_file.image_path())?;
+        let mut wasmtime = create_wasmtime(workdir, &deploy_file, self)?;
+
+        wasmtime.load_binaries(&mut image)?;
+
+        info!("Validation completed.");
+
+        Ok(())
+    }
+}
+
+/// Validates the deployed image.
+///
+/// Takes path to the workdir as an argument.
+pub fn start(workdir: impl AsRef<Path>) -> Result<()> {
+    RuntimeOptions::default().start(workdir)
```

Comment:
Is this method useful/needed at all given that the suggested way of starting an instance is via `RuntimeOptions::from_env().start()` now?

Reply:
this function is just for backward compatibility. old version does not use env.